connectivity/conn_http.h

```diff
@@ -23,31 +23,123 @@ extern "C" {
 #include "mbedtls/net.h"
 #include "mbedtls/ssl.h"
 
-#define HTTP_OK 200
-
+/**< Global pointer points to http resource body for passing response body to http_parser.
+ * This variable will be removed or changed to static pointer as soon as possible. Don't
+ * use this pointer directly.*/
 char *http_res_body;
 
 typedef struct {
-  bool https;
-
-  mbedtls_net_context *net_ctx;
-  mbedtls_entropy_context *entropy;
-  mbedtls_ctr_drbg_context *ctr_drbg;
-  mbedtls_ssl_context *ssl_ctx;
-  mbedtls_ssl_config *ssl_config;
-  mbedtls_x509_crt *cacert;
+  bool https;                         /**< Flag for check info has initialized or not */
+  mbedtls_net_context *net_ctx;       /**< mbedtls_next context                       */
+  mbedtls_entropy_context *entropy;   /**< mbedtls_entropy context                    */
+  mbedtls_ctr_drbg_context *ctr_drbg; /**< mbedlts_ctr_drbg context                   */
+  mbedtls_ssl_context *ssl_ctx;       /**< mbedtls_ssl context                        */
+  mbedtls_ssl_config *ssl_config;     /**< mbedtls_ssl_config context                 */
+  mbedtls_x509_crt *cacert;           /**< mbedtls_x509 container                     */
 } connect_info_t;
 
+/**
+ * @brief Open http connection
+ *
+ * @param[in, out] info Context for http connection
+ * @param[in] seed_nonce Personalization data, that is device-specific identifiers. Can be NULL.
+ * @param[in] host Http host to connect
+ * @param[in] port Http port to connect
+ *
+ * @return
+ * - #RET_HTTP_INIT failed on http init error
+ * - #RET_HTTP_CONNECT failed on http connect error
+ * - #RET_HTTP_CERT failed on http certificate setting error
+ * - #RET_HTTP_SSL failed on http ssl setting error
+ * - #RET_OK on success
+ * @see #retcode_t
+ */
 retcode_t http_open(connect_info_t *const info, char const *const seed_nonce, char const *const host,
                     char const *const port);
+
+/**
+ * @brief Send request to http connection
+ *
+ * @param[in] info Context for http connection
+ * @param[in] req  Buffer holding the data
+ *
+ * @return
+ * - #RET_OK on success
+ * - non-zero on error
+ * @see #retcode_t
+ */
 retcode_t http_send_request(connect_info_t *const info, const char *req);
+
+/**
+ * @brief Read response from http server
+ *
+ * @param[in] info Context for http connection
+ * @param[out] res Buffer that will hold the data
+ * @param[out] res_len Length of res
+ *
+ * @return
+ * - #RET_OK on success
+ * - non-zero on error
+ * @see #retcode_t
+ */
 retcode_t http_read_response(connect_info_t *const info, char *res, size_t res_len);
+
+/**
+ * @brief Close http connection
+ *
+ * @param[in] info Context for http connection
+ *
+ * @return
+ * - #RET_OK on success
+ * - non-zero on error
+ * @see #retcode_t
+ */
 retcode_t http_close(connect_info_t *const info);
 
+/**
+ * @brief Set post request message
+ *
+ * @param[in] api API for post request to https server, example: transaction/. Must be string.
```

Comment:
The parameter `api` is somehow misleading. Could you name it as `path`

Reply:
And revise the desctiption as `API path for POST request to http(s) server, i.e. "transaction/". It must be in string`